sdk/core/core/CMakeLists.txt

```diff
@@ -4,13 +4,32 @@
 cmake_minimum_required (VERSION 3.10)
 set(TARGET_NAME "az_core")
 
-project (${TARGET_NAME} LANGUAGES C)
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/version.txt)
+    file (STRINGS ${CMAKE_CURRENT_SOURCE_DIR}/version.txt VERSION_STRING)
+    message(STATUS "Read version from version.txt: " ${VERSION_STRING})
+    string(REGEX MATCHALL "([0-9]+).([0-9]+).([0.9]+)(.*)" PRE_RELEASE ${VERSION_STRING})
```

Comment:
Do we need the last `(.*)`? We likely will have hyphen after the major.minor.patch. For example, what about versions like 1.0.0-prelease.1?

Reply:
Yes, the everything else is required because it is not required to have '-'.  In fact when we release a GA version the version string will be something to the likes of "1.0.0".  In order to support prerelease and GA releases the capture can't expect a '-' to always be present